federation/server.c

```diff
@@ -0,0 +1,765 @@
+#include "c-spiffe/federation/server.h"
+#include "c-spiffe/logger/logger.h"
+#include "c-spiffe/spiffetls/spiffetls.h"
+#include "c-spiffe/utils/error.h"
+#include "c-spiffe/utils/picohttpparser.h"
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+spiffebundle_EndpointInfo *spiffebundle_EndpointInfo_New()
+{
+    spiffebundle_EndpointInfo *e_info = calloc(1, sizeof(*e_info));
+    mtx_init(&e_info->mutex, mtx_plain);
+    return e_info;
+}
+
+err_t spiffebundle_EndpointInfo_Free(spiffebundle_EndpointInfo *e_info)
+{
+    // logger_FmtPush(LOGGER_DEBUG, "spiffebundle_EndpointInfo_Free %p",
+    // e_info);
+    if(!e_info) {
+        // logger_Push(LOGGER_WARNING, "Error Freeing Endpoint (null)");
+        return ERR_NULL;
+    }
+    mtx_destroy(&e_info->mutex);
+    free(e_info);
+    return NO_ERROR;
+}
+
+spiffebundle_EndpointServer *spiffebundle_EndpointServer_New()
+{
+    // logger_FmtPush(LOGGER_DEBUG, "spiffebundle_EndpointServer_New %p");
+    spiffebundle_EndpointServer *new_server = calloc(1, sizeof(*new_server));
+
+    mtx_init(&new_server->mutex, mtx_plain);
+    mtx_lock(&new_server->mutex);
+    {
+        sh_new_strdup(new_server->endpoints);
+        sh_new_strdup(new_server->bundle_sources);
+        sh_new_strdup(new_server->bundle_tds);
+    }
+    mtx_unlock(&new_server->mutex);
```

Comment:
```suggestion
    sh_new_strdup(new_server->endpoints);
    sh_new_strdup(new_server->bundle_sources);
    sh_new_strdup(new_server->bundle_tds);
```
You don't need to lock the mutex when the variable was just created.

Reply:
fair enough